### DISCUSSION TITLE
What does wave execution mean ?

### DISCUSSION THREAD

**Discussion by yetingsky** [facebookincubator/velox#6290] [General]
Velox recently added wave execution primitives. What does wave execution mean?

**Comment by Yuhta** [comment]
Wave is a framework to run some operators on accelerators (GPU, FPGA, etc.).  It is still experimental, we are working on the CUDA implementation, and our partners will implement on other platforms.

**Reply by AlexanderChiuluvB** [reply]
Is there any plan on FPGA accelerations? If so, I'm interested

**Reply by Yuhta** [reply]
Intel will probably work on FPGA using their SYCL programming interface.  That being said, if you have different configurations you can always make implementations of your own hardware, it's designed to be an open extension point.

**Comment by aditi-pandit** [comment]
@Yuhta : Would you be able to provide more info on the wave code... How to run it etc ? Some folks from IBM research were curious about it.

**Reply by Yuhta** [reply]
With the current code case you can run the unit tests in that directory on an host with CUDA 11 or 12.  It's still a work in progress but some parts are runnable.  The E2E flow will be the same as a normal Prestissimo worker except running on host with CUDA (or other accelerators).
